### DISCUSSION TITLE
Xcode build example?

### DISCUSSION THREAD

**Discussion by KTRosenberg** [ggml-org/llama.cpp#486] [Q&A]
I would be interested in integrating llama directly into an Xcode project for macOS rather than as a command line application. Would anyone be able to advise on how to build the library that way? Alternatively, I imagine some interprocess communication or pipes could work.

**Comment by j-f1** [comment]
Give my branch at https://github.com/ggerganov/llama.cpp/pull/393 a shot, it adds SwiftPM support.

**Reply by KTRosenberg** [reply]
> Give my branch at #393 a shot, it adds SwiftPM support.

Oh great, I'll try!
I'm using C++ and C though (technically objective C). I have a bridging header to Swift though, and I can see it being easier to use Swift's package manager.
Looking at your swift example, how would I enter a string prompt? I see something's being loaded from a file. In the end, I'd probably like to be able to use an alpaca model.

Should I look at a particular example in the non-swift repo for this?

**Reply by j-f1** [reply]
The project is still written in C/C++ so you should still be able to import it as a library without using a bridging header (although I have never personally used SPM to do this).

You’d want to call `llama_tokenize` and pass the C string as a parameter to tokenize your prompt. The example code there is reduced from the source code of `main.cpp`.
